lib/config/index.js

```diff
@@ -74,6 +74,71 @@ exports.getLoaderConfig = function() {
   return cfg;
 };
 
+function updateNodelibTarget(target) {
+  target.registry = 'npm';
+  target.name = target.name.replace('github:jspm/nodelibs-', 'npm:jspm-nodelibs-');
+  target.package = target.package.replace('jspm/nodelibs-', 'jspm-nodelibs-');
+  target.exactPackage = target.exactPackage.replace('jspm/nodelibs-', 'jspm-nodelibs-');
+  target.exactName = target.exactName.replace('github:jspm/nodelibs-', 'npm:jspm-nodelibs-');
+  target.exactNameEncoded = target.exactNameEncoded.replace('github:jspm/nodelibs-', 'npm:jspm-nodelibs-');
+  target.setVersion('0.2.0');
+  return target;
+}
+
+function processNodelibs(loader, pjson) {
+  var installRequired = false;
+  Object.keys(loader.baseMap).forEach(function (key) {
+    var target = loader.baseMap[key];
+
+    if (target.name.substr(0, 21) == 'github:jspm/nodelibs-') {
+      target = updateNodelibTarget(target);
+      installRequired = true;
+    }
+  });
+
+  Object.keys(loader.depMap).forEach(function (key) {
+    var curMap = loader.depMap[key];
+
+    Object.keys(curMap).forEach(function (key) {
+      var target = curMap[key];
+
+      if (target.name.substr(0, 21) == 'github:jspm/nodelibs-') {
+        if (key == target.name.substr(21))
+          delete curMap[key];
+        else
+          installRequired = true;
```

Comment:
Alright. I wasn't aware of this distinction.
How do you propose to detect the difference between the beta upgrade path and the nodelibs upgrade path? I am not familiar enough with the code base to come up with something reasonable.